code/Controllers/CMSMain.php

```diff
@@ -350,6 +350,21 @@ public function LinkPageHistory()
         }
     }
 
+    /**
+     * Decide which is the active primary tab in a CMS edit screen
+     *
+     * @return string
+     */
+    public function getActiveTab()
+    {
+        if ($this instanceof CMSPageSettingsController) {
```

Comment:
Better coding style is for this method to return a literal string, and for those subclasses to override this to return the specific key.

Parent classes, as a principle, should not know about specific subclasses.